katsdpcontroller/scheduler.py

```diff
@@ -3278,6 +3297,34 @@ def depends_resources(cls, data):
             futures.append(asyncio.ensure_future(kill_one(node, kill_graph)))
         await asyncio.gather(*futures)
 
+    async def _reconciliation(self) -> None:
+        """Background task that periodically requests reconciliation.
+
+        See http://mesos.apache.org/documentation/latest/reconciliation/ for
+        a description of reconciliation.
+
+        We don't use the (somewhat complex) algorithm described there, instead
+        just periodically requesting either implicit or explicit
+        reconciliation (the former is useful to learn about tasks we thought
+        were dead, the latter to learn about tasks we thought were alive but
+        have been lost). This means that it may take somewhat longer to
+        converge.
+        """
+        explicit = True
+        while True:
+            try:
+                if explicit:
+                    tasks = [{'task_id': task_id} for task_id in self._active]
+                    logger.debug('Requesting explicit reconciliation of %d tasks', len(tasks))
+                else:
+                    tasks = []
+                    logger.debug('Requesting implicit reconciliation')
+                self._driver.reconcileTasks(tasks)
+            except Exception:
+                logger.warning('Exception during task reconciliation', exc_info=True)
```

Comment:
Why so general - what sort of failure modes to we expected from reconcileTasks ?

Reply:
I'm not expecting any failures, and I hope this will never run. But if one call to reconcileTasks fails, I want to keep trying rather than exiting the loop.